### PR TITLE
Home page editing

### PR DESCRIPTION
**Pull request by jonalport**
This work is dependent upon the CMS changes detailed here:
https://github.com/moneyadviceservice/cms/pull/278

The CMS changes can be released first without breaking frontend.

### CODE REVIEW

File: spec/lib/core/repository/cms/attribute_builder_spec.rb [hunk 1]
```diff
@@ -40,6 +40,14 @@ module Core::Repository::CMS
         expect(AttributeBuilder.build(response)['related_content']['next_link']).to be_present
       end
 
+      describe 'home page attributes' do
+        let(:body) { File.read('spec/fixtures/cms/home-page.json') }
+
+        it 'strips html tags' do
+          expect(AttributeBuilder.build(response)['hero_image']).to eql('http://localhost:3000/system/comfy/cms/files/files/000/000/067/original/atm_en.png')
```
Comment (benbarnett) [inline spec/lib/core/repository/cms/attribute_builder_spec.rb]: Is this exact URL correct?

**Comment by benbarnett** [comment]
Am I right in thinking that this both needs the CMS PR to be merged, but also the content needs to be populated? Looking at it this will just render what it finds but with no fallback if the data is empty?

**Comment by jonalport** [comment]
@benbarnett: Correct, the CMS changes must be released and the content published, before this is released. Details are in the CMS PR.

**Comment by asmega** [comment]
ready for review

File: lib/core/repository/cms/attribute_builder.rb [hunk 2]
```diff
@@ -6,6 +6,22 @@ def self.build(response)
...
+        attributes.select {|k,v| k.match(/_(\d+)_/) }.each do |k,v|
+          _, object, number, field = k.match(/(\w+)_(\d+)_(\w+)/).to_a
+          number = number.to_i - 1
+          key = object.pluralize
+
+          attributes[key] ||= []
+          attributes[key][number] ||= {}
+          attributes[key][number][field] = v
+        end
+
```
Comment (tomas-stefano) [inline lib/core/repository/cms/attribute_builder.rb]: I know that it is just assignments, but can you move this to private methods, please?

**Comment by asmega** [comment]
@tomas-stefano @darokel I've refactored attribute builder and squashed the commits

**Comment by tomas-stefano** [comment]
LGTM :+1:

**Comment by darokel** [comment]
👍🏼

**Comment by tomas-stefano** [comment]
:+1:

**Comment by tomas-stefano** [comment]
:+1:

**Comment by darokel** [comment]
All good :+1:

**Pull request merged**